src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1876,10 +1877,10 @@ private BoundExpression BindCast(CastExpressionSyntax node, DiagnosticBag diagno
                 return BindExplicitNullableCastFromNonNullable(node, operand, targetType, diagnostics);
             }
 
-            return BindCastCore(node, operand, targetType, wasCompilerGenerated: operand.WasCompilerGenerated, diagnostics: diagnostics);
+            return BindCastCore(node, operand, targetType, wasCompilerGenerated: operand.WasCompilerGenerated, diagnostics: diagnostics, isNullable: targetTypeWithNullability.IsNullable);
```

Comment:
>IsNullable [](start = 175, length = 10)

Does it matter if the type is a nullable value type?

Reply:
No, it shouldn't matter.

---
In reply to: [175652551](https://github.com/dotnet/roslyn/pull/25516#discussion_r175652551) [](ancestors = 175652551)